### DISCUSSION TITLE
Added: LDM notes for April 6th and April 8th, 2020

### DISCUSSION THREAD

**Discussion by agocke** [dotnet/csharplang#3347] [General]
## April 8, 2020

[C# Language Design Notes for April 8, 2020](https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-08.md)

1. `e is dynamic` pure null check
2. Target typing `?:`
3. Inferred type of an `or` pattern
4. Module initializers

## April 6, 2020

[C# Language Design Notes for April 6, 2020](https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-06.md)

1. Record Monday: Init-only members

**Comment by jnm2** [comment]
@RikkiGibson The prototype link in https://github.com/RikkiGibson/csharplang/blob/module-initializers/proposals/module-initializers.md is broken. It should be https://github.com/jnm2/roslyn/tree/module_initializer.

**Comment by RikkiGibson** [comment]
Thanks, updated!

**Comment by jnm2** [comment]
https://github.com/dotnet/csharplang/blob/master/meetings/2020/LDM-2020-04-08.md#module-initializers

> #### Conclusion
>
> Let's let any static method be a module initializer, and mark that method using a well-known attribute. We'll also allow multiple module initializer methods, and they will each be called in a reserved, but deterministic order.

I like this a lot.

**Comment by SingleAccretion** [comment]
What would be the appropriate issue to comment on `init` syntax?

**Comment by iam3yal** [comment]
@SingleAccretion Don't know whether there is a specific discussion about it but you're free to create one, worst-case scenario it would be a dup.

**Comment by MgSam** [comment]
I didn't see it addressed in the design notes or the proposal- for properties, why has the design moved from `init` being used in place of the word `set` to being a modifier on `set`? I prefer the terser version where its used in place of `set`.

Otherwise, I love the `init` proposal. It is a syntax-light way of adding substantial new functionality with broad usage scenarios to C#. I think it likely is more broadly useful than even records are.

**Comment by SingleAccretion** [comment]
@MgSam Presumably to keep it a "modifier" keyword, as it can also be applied to fields.

To be honest, I am skeptical about `init` (not the concept, but the word itself). It feels out of place for C#. I think I am going to open an issue about it as soon as I have enough time to properly analyze the alternatives.

**Comment by qrli** [comment]
On `init` syntax, I feel reusing `readonly` can be an option:
```cs
    public readonly int P { get; set; }
```
It feels similar to readonly fields, which can be set at initialization, but not later.

**Comment by SingleAccretion** [comment]
@qrli This would not work for fields (which enable `init _value;` `Value { init set; }` pattern).

**Comment by 333fred** [comment]
Also, `readonly` can already be applied to getters and setters and means something completely different.
